Robust.Shared/Prototypes/MultiRootInheritanceGraph.cs

```diff
@@ -38,6 +38,9 @@ public void Add(T id, params T[] parents)
         //check for circular inheritance
         foreach (var parent in parents)
         {
+            if (EqualityComparer<T>.Default.Equals(parent,id))
+                throw new InvalidOperationException($"Circular Inheritance detected for id \"{id}\" and parent \"{parent}\"");
```

Comment:
```suggestion
                throw new InvalidOperationException($"Self Inheritance detected for id \"{id}\"");
```
Might as well make the message distinct